urllib3/connectionpool.py

```diff
@@ -383,8 +383,13 @@ def _make_request(self, conn, method, url, timeout=_Default, chunked=False,
         try:
             try:  # Python 2.7, use buffering of HTTP responses
                 httplib_response = conn.getresponse(buffering=True)
-            except TypeError:  # Python 2.6 and older
-                httplib_response = conn.getresponse()
+            except TypeError:  # Python 2.6 and older, Python 3
+                try:
+                    httplib_response = conn.getresponse()
+                except Exception as e:
```

Comment:
The `e` here is not the `TypeError`. I am not masking `e`, I just make it not chain to the `TypeError` but start a new chain instead.